src/main/kotlin/com/cognifide/gradle/aem/instance/action/DownAction.kt

```diff
@@ -0,0 +1,71 @@
+package com.cognifide.gradle.aem.instance.action
+
+import com.cognifide.gradle.aem.instance.*
+import com.cognifide.gradle.aem.internal.Behaviors
+import com.cognifide.gradle.aem.internal.ProgressLogger
+import org.gradle.api.Project
+
+class DownAction(project: Project, val instances: List<Instance>) : AbstractAction(project) {
+
+    var stableState = config.awaitStableCheck
+
+    override fun perform() {
+        if (instances.isEmpty()) {
+            logger.info("No instances to shutdown.")
+            return
+        }
+
+        shutDownInstances()
+    }
+
+    private fun shutDownInstances() {
+        val handles = Instance.handles(project)
```

Comment:
convert constructor val `instances` into handles here, it is a parameter of that action potentially reusable in Gradle build script / think of it while implementing